src/activities/morse_code/MorseCode.qml

```diff
@@ -86,8 +86,15 @@ ActivityBase {
             }
 
             function initSubLevel() {
+                if(dataset[currentLevel].values[0] == '_random_') {
+                    var randomIndex = Math.floor(Math.random() * dataset[currentLevel].values[1].length)
+                    questionValue = dataset[currentLevel].values[1][randomIndex]
+                    dataset[currentLevel].values[1].splice(randomIndex,1)
+                    score.numberOfSubLevels = dataset[currentLevel].values[2]
```

Comment:
You should only compute once the number of sublevels for a level in initLevel. In this case, you can always use the list length